### PR TITLE
feat: revamp GetFileStats to use internal grpc comm

### PR DESCRIPTION
**Pull request by kushthedude**

### CODE REVIEW

**Comment by cnlangzi** [comment]
@kushthedude  please resolve conflicting files first. thanks.

File: code/go/0chain.net/blobbercore/handler/storage_handler.go [hunk 1]
```diff
@@ -325,56 +323,83 @@ func (fsh *StorageHandler) AddCollaborator(ctx context.Context, r *http.Request)
 	return result, nil
 }
 
-func (fsh *StorageHandler) GetFileStats(ctx context.Context, r *http.Request) (interface{}, error) {
-	if r.Method == "GET" {
-		return nil, common.NewError("invalid_method", "Invalid method used. Use POST instead")
-	}
-	allocationTx := ctx.Value(constants.ALLOCATION_CONTEXT_KEY).(string)
+func (fsh *StorageHandler) GetFileStats(ctx context.Context, request *blobbergrpc.GetFileStatsRequest) (interface{}, error) {
```
Comment (cnlangzi) [inline code/go/0chain.net/blobbercore/handler/storage_handler.go:326]: why we remove "GET" checking?

**Comment by kushthedude** [comment]
This is not going into the master.

File: code/go/0chain.net/blobbercore/handler/storage_handler.go [hunk 2]
```diff
@@ -325,56 +323,83 @@ func (fsh *StorageHandler) AddCollaborator(ctx context.Context, r *http.Request)
...
-		return nil, common.NewError("invalid_signature", "Invalid signature")
+	if err != nil || !valid {
+		return nil, errors.Wrap(err, "invalid signature passed in request")
 	}
 
 	clientID := ctx.Value(constants.CLIENT_CONTEXT_KEY).(string)
-	if len(clientID) == 0 || allocationObj.OwnerID != clientID {
-		return nil, common.NewError("invalid_operation", "Operation needs to be performed by the owner of the allocation")
+	if (clientID == "") || (allocationObj.OwnerID != clientID) {
+		return nil, errors.Wrap(errors.New("Invalid Operation"), "operation needs to be performed by allocation's owner")
```
Comment (cnlangzi) [inline code/go/0chain.net/blobbercore/handler/storage_handler.go:348]: it is better to move "InvalidOperation" to a variable if we want to wrap it.

e.g

```
var ErrInvalidOperation = errors.New("Invalid Operation")
```

File: code/go/0chain.net/blobbercore/handler/storage_handler.go [hunk 3]
```diff
@@ -325,56 +323,83 @@ func (fsh *StorageHandler) AddCollaborator(ctx context.Context, r *http.Request)
...
-		return nil, common.NewError("invalid_parameters", "Invalid file path. "+err.Error())
+		Logger.Error("Invalid file path passed in the request")
+		return nil, errors.Wrapf(err, "invalid file path")
 	}
 
+	// additional check to ensure the FILE
 	if fileref.Type != reference.FILE {
-		return nil, common.NewError("invalid_parameters", "Path is not a file.")
+		Logger.Error("Path is not a file")
+		return nil, errors.Wrapf(errors.New("Invalid Parameters"), "path is not a file")
```
Comment (cnlangzi) [inline code/go/0chain.net/blobbercore/handler/storage_handler.go:369]: same as `ErrInvalidOperator`. please move those errors to variables
```
var (
 ErrInvalidOperation = errors.New("Invalid Operation")
 ErrInvalidParameters = errorw.New("Invalid Parameters")
....
 

```

File: code/go/0chain.net/blobbercore/handler/handler_test.go [hunk 4]
```diff
@@ -546,77 +546,77 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
...
-				mock.ExpectQuery(regexp.QuoteMeta(`SELECT * FROM "write_markers"`)).
-					WithArgs(sqlmock.AnyArg()).
-					WillReturnRows(
-						sqlmock.NewRows([]string{}).
-							AddRow(),
-					)
-			},
-			wantCode: http.StatusOK,
-		},
+		//{
```
Comment (cnlangzi) [inline code/go/0chain.net/blobbercore/handler/handler_test.go:549]: please remove them if they are unused .

File: code/go/0chain.net/blobbercore/handler/handler_test.go [hunk 5]
```diff
@@ -151,14 +151,14 @@ func setupHandlers() (*mux.Router, map[string]string) {
...
 
-	sPath := "/v1/file/stats/{allocation}"
-	sName := "Stats"
-	router.HandleFunc(sPath, common.UserRateLimit(
-		common.ToJSONResponse(
-			WithReadOnlyConnection(FileStatsHandler),
-		),
-	),
-	).Name(sName)
+	//sPath := "/v1/file/stats/{allocation}"
```
Comment (cnlangzi) [inline code/go/0chain.net/blobbercore/handler/handler_test.go:154]: please remove them if they are unused

File: code/go/0chain.net/blobbercore/handler/handler.go [hunk 6]
```diff
@@ -53,7 +53,7 @@ func SetupHandlers(r *mux.Router) {
 	//object info related apis
 	//r.HandleFunc("/allocation", common.UserRateLimit(common.ToJSONResponse(WithConnection(AllocationHandler))))
 	//r.HandleFunc("/v1/file/meta/{allocation}", common.UserRateLimit(common.ToJSONResponse(WithReadOnlyConnection(FileMetaHandler))))
-	r.HandleFunc("/v1/file/stats/{allocation}", common.UserRateLimit(common.ToJSONResponse(WithReadOnlyConnection(FileStatsHandler))))
+	//r.HandleFunc("/v1/file/stats/{allocation}", common.UserRateLimit(common.ToJSONResponse(WithReadOnlyConnection(FileStatsHandler))))
```
Comment (cnlangzi) [inline code/go/0chain.net/blobbercore/handler/handler.go:56]: is it safe to remove?  I know it is still used on gosdk. https://github.com/0chain/gosdk/blob/826e79c0eb2030c6ba1ada3f0aa7928c67e5d590/zboxcore/zboxutil/http.go#L326

File: code/go/0chain.net/blobbercore/handler/handler_test.go (hunk 5 above)
Comment (cnlangzi) [reply code/go/0chain.net/blobbercore/handler/handler_test.go:154]: the api is still used on gosdk.

**Review by cnlangzi — CHANGES_REQUESTED** [review]

**Comment by cnlangzi** [comment]
> This is not going into the master.

OK

**Comment by cnlangzi** [comment]
@kushthedude could you convert it `draft` ?

File: code/go/0chain.net/blobbercore/handler/storage_handler.go (hunk 1 above)
Comment (kushthedude) [reply code/go/0chain.net/blobbercore/handler/storage_handler.go:326]: Done by grpc-gateway!

File: code/go/0chain.net/blobbercore/handler/storage_handler.go (hunk 2 above)
Comment (kushthedude) [reply code/go/0chain.net/blobbercore/handler/storage_handler.go:348]: This is the next step when complete error package will be deprecated, as it is only duplicated code

File: code/go/0chain.net/blobbercore/handler/storage_handler.go (hunk 3 above)
Comment (kushthedude) [reply code/go/0chain.net/blobbercore/handler/storage_handler.go:369]: Answered above.

File: code/go/0chain.net/blobbercore/handler/handler_test.go (hunk 4 above)
Comment (kushthedude) [reply code/go/0chain.net/blobbercore/handler/handler_test.go:549]: Backward compatibility, won't remove untill deployed to master

File: code/go/0chain.net/blobbercore/handler/handler_test.go (hunk 5 above)
Comment (kushthedude) [reply code/go/0chain.net/blobbercore/handler/handler_test.go:154]: Read about grpc-gateway please

**Comment by kushthedude** [comment]
> @kushthedude could you convert it `draft` ?

See the  the grpc channel, the tests are buggy for the following endpoint hence on hold.

File: code/go/0chain.net/blobbercore/handler/handler.go (hunk 6 above)
Comment (kushthedude) [reply code/go/0chain.net/blobbercore/handler/handler.go:56]: see above

**Pull request closed without merging**